include/cifer/data/mat.h

```diff
@@ -173,4 +174,26 @@ void cfe_mat_from_vec(cfe_mat *m, cfe_vec *v);
  */
 void cfe_mat_transpose(cfe_mat *t, cfe_mat *m);
 
+/**
+ * Returns a matrix obtained from m by removing row i and column j.
+ * It returns an error if i >= number of rows of m, or if j >= number of
+ * columns of m.
+ */
+void cfe_mat_extract_submatrix(cfe_mat *m, cfe_mat *min, size_t i, size_t j);
+
+/**
+ * Determinant of a square matrix.
+ */
+void cfe_mat_determinant(cfe_mat *m, mpz_t det);
+
+/**
+ * Inverse of a matrix over a modular field Z_mod.
+ */
+cfe_error cfe_mat_inverse_mod(cfe_mat *m, cfe_mat *inverse_mat, mpz_t mod);
```

Comment:
Can you change the order of parameters so that the result is always first?

Reply:
Yes, I will change it also in the extraction and determinant function.